lib/ansible/plugins/connection/__init__.py

```diff
@@ -285,6 +285,24 @@ def update_vars(self, variables: dict[str, t.Any]) -> None:
                 display.debug('Set connection var {0} to {1}'.format(varname, value))
                 variables[varname] = value
 
+    # TODO find better name and place for this. Public API? On AnsiblePlugin?
+    def _var_options(self, variables, templar):
+        # create dict of 'templated vars'
+        var_options = {
+            '_extras': {},
+        }
+        for var_name in C.config.get_plugin_vars('connection', self._load_name):
+            if var_name in variables:
+                var_options[var_name] = templar.template(variables[var_name])
+
+        # add extras if plugin supports them
+        if getattr(self, 'allow_extras', False):
+            for var_name in variables:
+                if var_name.startswith(f'ansible_{self.extras_prefix}_') and var_name not in var_options:
+                    var_options['_extras'][var_name] = templar.template(variables[var_name])
```

Comment:
Moving this to common code for consistency between the two paths makes sense, although I hope extra vars for connection plugins will be deprecated/phased out eventually, since now there's a way to document the variables, which also doubles as the argument spec and provides a consistent UX.